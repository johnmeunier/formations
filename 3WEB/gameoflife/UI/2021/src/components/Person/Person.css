.person {
  background-color: #fff;
  border: 5px solid #942911;
  color: #942911;
  width: 25%;
  margin: 15px 0px 15px 0px;
  padding: 15px;
}

.contact {
  display: flex;
  align-items: center;
}

.contact__icon {
  width: 20px;
  margin-right: 5px;
}
