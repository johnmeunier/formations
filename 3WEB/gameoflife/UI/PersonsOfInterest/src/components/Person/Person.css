.card {
  background-color: #fff;
  border: 3px solid #942911;
  width: 30%;
  margin: 15px 0px 15px 0px;
  padding: 15px;
}

.card__name {
  color: #942911;
  font-weight: 700;
  font-size: 1.4rem;
  text-align: center;
}
.contacts {
  margin: 10px 0px;
}

.contact {
  display: flex;
  align-items: center;
}

.contact__icon {
  width: 20px;
  margin-right: 5px;
}
