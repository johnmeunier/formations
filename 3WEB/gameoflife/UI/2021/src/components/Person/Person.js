import "./Person.css";
import mailIcon from "./mail-icon.png";

export const Person = ({ name, email, phone, greeting, gender }) => (
  <div className="person">
    <h3 className="person__name">
      {gender === "male" ? <span>&#9794;</span> : <span>&#9792;</span>}
      {name}
    </h3>
    <div className="contacts">
      <h4 className="contact">
        <img src={mailIcon} className="contact__icon" />
        <a href={`mailto:${email}`}>{email}</a>
      </h4>
      <h4 className="contact">
        <a href={`tel:${phone}`}>{phone}</a>
      </h4>
    </div>
    <p>{greeting}</p>
  </div>
);
