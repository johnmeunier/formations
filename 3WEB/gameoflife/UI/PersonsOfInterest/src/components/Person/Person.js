import { Link } from "react-router-dom";

import "./Person.css";
import mailIcon from "./mail-icon.png";

export const Person = ({ name, email, phone, greeting, gender, guid }) => (
  <div className="card">
    <h3 className="card__name">
      {gender === "male" ? <span>&#9794;</span> : <span>&#9792;</span>}
      {name}
    </h3>
    <div className="contacts">
      <h4 className="contact">
        <img src={mailIcon} className="contact__icon" />
        <a href={`mailto:${email}`}>{email}</a>
      </h4>
      <h4 className="contact">
        <a href={`tel:${phone}`}>{phone}</a>
      </h4>
    </div>
    <p>{greeting}</p>
    <Link to={`/details/${guid}`}>Voir le d√©tails de {name}</Link>
  </div>
);
