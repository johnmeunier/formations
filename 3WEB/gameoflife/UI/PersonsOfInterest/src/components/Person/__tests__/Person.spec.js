import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import { Person } from "../Person";

describe("snapshot", () => {
  const props = {
    name: "testName",
    email: "testMail@toto.com",
    phone: "+33333",
    greeting: "Hello",
    gender: "M",
    guid: "12KLMSDJKLMQSD",
  };
  it("should render correctly", () => {
    const { container } = render(
      <Router>
        <Person {...props} />
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should render correctly with a bad email", () => {
    const { container } = render(
      <Router>
        <Person {...props} email="sdmlfmsdf" />
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
