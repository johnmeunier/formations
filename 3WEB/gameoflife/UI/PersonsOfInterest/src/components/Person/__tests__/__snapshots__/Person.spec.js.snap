// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should render correctly 1`] = `
<div
  class="card"
>
  <h3
    class="card__name"
  >
    <span>
      ♀
    </span>
    testName
  </h3>
  <div
    class="contacts"
  >
    <h4
      class="contact"
    >
      <img
        class="contact__icon"
        src="mail-icon.png"
      />
      <a
        href="mailto:testMail@toto.com"
      >
        testMail@toto.com
      </a>
    </h4>
    <h4
      class="contact"
    >
      <a
        href="tel:+33333"
      >
        +33333
      </a>
    </h4>
  </div>
  <p>
    Hello
  </p>
  <a
    href="/details/12KLMSDJKLMQSD"
  >
    Voir le détails de 
    testName
  </a>
</div>
`;

exports[`snapshot should render correctly with a bad email 1`] = `
<div
  class="card"
>
  <h3
    class="card__name"
  >
    <span>
      ♀
    </span>
    testName
  </h3>
  <div
    class="contacts"
  >
    <h4
      class="contact"
    >
      <img
        class="contact__icon"
        src="mail-icon.png"
      />
      <a
        href="mailto:sdmlfmsdf"
      >
        sdmlfmsdf
      </a>
    </h4>
    <h4
      class="contact"
    >
      <a
        href="tel:+33333"
      >
        +33333
      </a>
    </h4>
  </div>
  <p>
    Hello
  </p>
  <a
    href="/details/12KLMSDJKLMQSD"
  >
    Voir le détails de 
    testName
  </a>
</div>
`;
